### PR TITLE
MAINT - Pre release docs and improvements

### PR DESCRIPTION
**Pull request by trallard**
## Description

THis PR adds miscellaneous improvements to prepare the repo for a release:

- :pushpin: Pin Docker base images
- :wrench: Update pyproject.toml
- :memo: Update release docs
- :construction_worker: Update GH actions
- :memo: Finish docs

## Pull request checklist

- [x] Did you test this change locally?
- [x] Did you update the documentation (if required)?
- [ ] Did you add/update relevant tests for this change (if required)?

### CODE REVIEW

File: docusaurus-docs/community/release.md [hunk 1]
```diff
@@ -3,3 +3,108 @@ description: Release process
...
+:::warning
+[jupyterlab-conda-store](https://github.com/conda-incubator/jupyterlab-conda-store) has a direct dependency on [conda-store-ui](https://github.com/conda-incubator/conda-store-ui). Make sure to release conda-store-ui before releasing jupyterlab-conda-store.
+:::
+
+### Release checklist (conda-store)
+
+Create an issue and copy & paste the steps below to release a new conda-store version. Close the issue when it is done.
+
+:::caution
+There are two packages the [conda-store](https://github.com/conda-incubator/conda-store) repository; [`conda-store`](https://github.com/conda-incubator/conda-store/tree/main/conda-store) and [`conda-store-server`](https://github.com/conda-incubator/conda-store/tree/main/conda-store-server). Make sure to update both packages when releasing a new version.
```
Comment (pavithraes) [inline docusaurus-docs/community/release.md:59]:
```suggestion
There are two packages in the [conda-store](https://github.com/conda-incubator/conda-store) repository; [`conda-store`](https://github.com/conda-incubator/conda-store/tree/main/conda-store) and [`conda-store-server`](https://github.com/conda-incubator/conda-store/tree/main/conda-store-server). Make sure to update both packages when releasing a new version.
```

**Review by pavithraes — APPROVED** [review]
Thanks, @trallard! This looks good to me!

Only one tiny typo correction, so I'll approve and you can merge when ready. :)

**Review by kcpevey — APPROVED** [review]
LGTM from content/grammar. I didn't run anything though since it seemed unnecessary.

File: docusaurus-docs/community/release.md [hunk 2]
```diff
@@ -3,3 +3,108 @@ description: Release process
...
+:::
+
+```md
```
[the remark this answers is not in the document]
Comment (trallard) [reply docusaurus-docs/community/release.md:62]: Sure we can do that in a separate PR, in the meantime I will merge this as is

**Pull request merged**